src/bloqade/squin/op/number.py

```diff
@@ -0,0 +1,5 @@
+import numbers
+
+from kirin.ir.attrs.types import PyClass
+
+Number = PyClass(numbers.Number)
```

Comment:
```suggestion
NumberType = PyClass(numbers.Number)
```
Can we call `NumberType` to be consistent?